tests/kafkatest/services/kafka.py

```diff
@@ -199,29 +208,53 @@ def restart_node(self, node, wait_sec=0, clean_shutdown=True):
     def leader(self, topic, partition=0):
         """ Get the leader replica for the given topic and partition.
         """
-        cmd = "/opt/kafka/bin/kafka-run-class.sh kafka.tools.ZooKeeperMainWrapper -server %s " \
-              % self.zk.connect_setting()
-        cmd += "get /brokers/topics/%s/partitions/%d/state" % (topic, partition)
-        self.logger.debug(cmd)
+        topic_partition_data = self.zk_get_data("/brokers/topics/%s/partitions/%d/state" % (topic, partition))
```

Comment:
That's part of why I was asking about the zkconnect setting. It's just something to keep in mind because we're increasing the number of ways we're relying on the semantics of the ZookeeperService class, and I realized that chroots are something we may want to test at some point. May not require anything in this patch, could be a parameter where we only use the default, or something else. It's worth thinking through the implications as we add more utility methods like this.